lib/iris/_lazy_data.py

```diff
@@ -47,7 +47,14 @@ def is_lazy_data(data):
     return result
 
 
-def _optimum_chunksize(chunks, shape, limit=None, dtype=np.dtype("f4")):
+@lru_cache
+def _optimum_chunksize_internals(
+    chunks,
+    shape,
+    limit=None,
```

Comment:
do we need `dask_array_chunksize` or will `limit` suffice?

Reply:
I think we need `dask_array_chunksize` so that everything external that the function needs is in the arguments, unless we calculate limit in the wrapper using the logic inside the internals (but that seemed like more changes, and breaks up the logic so it's harder to follow)?

Reply:
It also needs to go into the docstring